lib/retry.go

```diff
@@ -0,0 +1,20 @@
+package lib
+
+import (
+	"context"
+	"errors"
+	"time"
+)
+
+func WithRetries(ctx context.Context, retryFn func() error, options struct {
+	retriesCount    int
+	retriesInterval int
+}) error {
```

Comment:
É legal exportarmos um tipo pra esse `struct`. Se não, vai ser preciso declarar um struct anônimo do lado da aplicação no momento da chamada. Além disso, o intervalo de retentativa (`retriesInterval`) já pode ser recebido como uma duration:
```suggestion
type RetryOptions struct {
	RetriesCount    int
	RetriesInterval time.Duration
}

func WithRetries(ctx context.Context, retryFn func() error, options RetryOptions) error {
```